### PR TITLE
Use revisions instead of nll compare mode for `/self/` ui tests

### PR DESCRIPTION
**Pull request by marmeladema**
r? @jackh726

### CODE REVIEW

File: src/test/ui/self/arbitrary_self_types_pin_lifetime_mismatch.base.stderr [hunk 1]
```diff
@@ -13,7 +13,7 @@ LL |     fn a<'a>(self: Pin<&'a Foo>, f: &'a Foo) -> &Foo { f }
    |         ++++            ++           ++
 
 error[E0623]: lifetime mismatch
-  --> $DIR/arbitrary_self_types_pin_lifetime_mismatch.rs:8:76
```
Comment (jackh726) [inline src/test/ui/self/arbitrary_self_types_pin_lifetime_mismatch.base.stderr:16]: We lose the suggestion above. Can you file an issue for this?

**Review by jackh726 — APPROVED** [review]
LGTM, other than needing to file an issue for the note and suggestion (it happens in several of these tests).

r=me after that

File: src/test/ui/self/arbitrary_self_types_pin_lifetime_mismatch.base.stderr (hunk 1 above)
Comment (marmeladema) [reply src/test/ui/self/arbitrary_self_types_pin_lifetime_mismatch.base.stderr:16]: Filed https://github.com/rust-lang/rust/issues/96157

**Comment by marmeladema** [comment]
@jackh726 done!

**Pull request merged**